src/publishers/serial_publisher.jl

```diff
@@ -7,8 +7,8 @@ Write data over a serial port.
 """
 mutable struct SerialPublisher <: Publisher
     serial_port::LibSerialPort.SerialPort
-
     name::String
+    has_published::Threads.Atomic{Bool}
```

Comment:
Is the `has_published` flag for later use to check to see if the publisher is stale? Also is it worth putting this inside another struct similar to `SubscriberFlags` and require all publisher types have a `getflags()` defined?

Reply:
Yep, it's just a check to verify that the `publish` method has actually been called (presumably by the node, as reported by `printstatus`. That's a good idea to mirror the flags struct in the subscriber. I just stuck it here as a quick fix.

Reply:
I made a `PublisherFlags` struct and a `getflags` method on `Publisher`. I implemented it on `ZmqSubscriber` but not `SerialSubscriber`. I'll leave that for you in the next PR.